strawberry/channels/handlers/graphql_transport_ws_handler.py

```diff
@@ -0,0 +1,58 @@
+from datetime import timedelta
+from typing import Any, Optional
+
+from strawberry.channels.handlers.base import ChannelsWSConsumer
+from strawberry.schema import BaseSchema
+from strawberry.subscriptions import GRAPHQL_TRANSPORT_WS_PROTOCOL
+from strawberry.subscriptions.protocols.graphql_transport_ws.handlers import (
+    BaseGraphQLTransportWSHandler,
+)
+
+
+class GraphQLTransportWSHandler(BaseGraphQLTransportWSHandler):
+    def __init__(
+        self,
+        schema: BaseSchema,
+        debug: bool,
+        connection_init_wait_timeout: timedelta,
+        get_context,
+        get_root_value,
+        ws: ChannelsWSConsumer,
+    ):
+        super().__init__(schema, debug, connection_init_wait_timeout)
+        self._get_context = get_context
+        self._get_root_value = get_root_value
+        self._ws = ws
+
+    async def get_context(self) -> Any:
+        return await self._get_context(request=self._ws)
+
+    async def get_root_value(self) -> Any:
+        return await self._get_root_value(request=self._ws)
+
+    async def send_json(self, data: dict) -> None:
+        await self._ws.send_json(data)
+
+    async def close(self, code: int = 1000, reason: Optional[str] = None) -> None:
+        # FIXME: We are using `self._ws.base_send` directly instead of `self._ws.close`
+        # because the later doesn't accept the `reason` argument.
+        await self._ws.base_send(
+            {
+                "type": "websocket.close",
+                "code": code,
+                "reason": reason or "",
+            }
+        )
+
+    async def handle_request(self) -> Any:
+        await self._ws.accept(subprotocol=GRAPHQL_TRANSPORT_WS_PROTOCOL)
+
+    async def handle_disconnect(self, code):
+        timeout_task = self.connection_init_timeout_task
+        if timeout_task and not timeout_task.done() and not timeout_task.cancelled():
+            timeout_task.cancel()
+
+        for operation_id in list(self.subscriptions.keys()):
```

Comment:
```suggestion
        for operation_id in self.subscriptions.keys():
```

we don't need to convert to a list, right?

Reply:
I agree. I noticed that this code was copied from other implementations (more specifically, asgi and aiohttp) which does convert to a list.

Is there a reason to why they do that? I know that this kind of code is usually done to be able to modify the dict in the loop, because a dict should not be modified while iterating over it.

@DoctorJohn do you know if there's a reason for this? Or is it safe to just iterate without making a copy of the keys?

Reply:
Should work without the cast. Looks like something I missed to change after introducing the shared base subscription classes.

Reply:
I tried removing and got this error from tests lol:
```
ERROR 😡  tests/channels/test_graphql_transport_ws.py::test_duplicated_operation_ids - RuntimeError: dictionary changed size during iteration
```

Reply:
Oh that error sounds familiar. Pretty sure that's why I "copied" the keys. The dict size can change while we await `cleanup_operation`.

Reply:
Should we consider this comment to be "resolved" then?